packages/calypso-e2e/src/lib/pages/site-editor-page.ts

```diff
@@ -0,0 +1,170 @@
+import { Locator, Page } from 'playwright';
+import {
+	BlockInserter,
+	EditorGutenbergComponent,
+	EditorSidebarBlockInserterComponent,
+	EditorToolbarComponent,
+	EditorWelcomeTourComponent,
+} from '..';
+import { getCalypsoURL } from '../../data-helper';
+import envVariables from '../../env-variables';
+
+const wpAdminPath = 'wp-admin/themes.php';
+
+const selectors = {
+	editorIframe: `iframe.is-loaded[src*="${ wpAdminPath }"]`,
+	editorRoot: 'body.block-editor-page',
+	editorCanvasIframe: 'iframe[name="editor-canvas"]',
+	editorCanvasRoot: '.wp-site-blocks',
+	templateLoadingSpinner: '[aria-label="Block: Template Part"] .components-spinner',
+};
+
+/**
+ * Represents an instance of the FSE site editor.
+ * This class is composed of editor components, combining them into larger flows.
+ */
+export class SiteEditorPage {
+	private page: Page;
+
+	private editorToolbarComponent: EditorToolbarComponent;
+	private editorGutenbergComponent: EditorGutenbergComponent;
+	private editorSidebarBlockInserterComponent: EditorSidebarBlockInserterComponent;
+	private editorWelcomeTourComponent: EditorWelcomeTourComponent;
+
+	/**
+	 * Constructs an instance of the page POM class.
+	 *
+	 * @param {Page} page The underlying page.
+	 */
+	constructor( page: Page ) {
```

Comment:
Because there's not a 1:1 correlation between whether we're testing on atomic and whether this is gutenframed, I opted out the `target` constructor param, and made the frame safety based on runtime URL.

Because the lookup happens in a property function, it should happen right at the exact time of someone doing selecting, so we can trust the URL!

Reply:
Interesting. So if I understand correctly, the FSE editor page is still different depending on Simple/Atomic, but the way it loads means that the page is guaranteed to be fully loaded?

Reply:
>the FSE editor page is still different depending on Simple/Atomic
I actually am not sure this is completely true. From what I can tell (but my understanding is still somewhat limited), it looks like you can end up with both states (Gutenframed and not) on both Simple _and_ Atomic sites. 

>but the way it loads means that the page is guaranteed to be fully loaded
Actually, I meant by the fact that `editor` was defined in a property, it means it is looked up in real-time. We don't check `page.url()` until you actually call a method on the `SiteEditorPage` (as opposed to during construction, which could happen at any point in a script), which you presumably wouldn't be doing until you've ensured it has finished loading.

Reply:
> it looks like you can end up with both states (Gutenframed and not) on both Simple and Atomic sites.

💩 

> which you presumably wouldn't be doing until you've ensured it has finished loading

Ah okay, so something like guarantees that we're loaded:

```
const siteEditorPage = new SiteEditorPage(page);
await siteEditorPage.waitUntilLoaded();
await siteEditorPage.someOtherMethod();
```

By calling `waitUntilLoaded` we've guaranteed the page is loaded so the URL can be checked to use the appropriate top level editor selector.

Does that sound right?

Reply:
>By calling waitUntilLoaded we've guaranteed the page is loaded so the URL can be checked to use the appropriate top level editor selector. Does that sound right?

Yes! 🎉 

In that example you gave, if we checked the URL in the constructor, the URL could be wrong. But by doing it real time in the actual action method, we know we're actually in the editor and ready to roll! 👍